examples/orgchart.rs

```diff
@@ -0,0 +1,141 @@
+//!
+//! This is similar to the motivating example in chapter 1 of the Differential
+//! Dataflow book at <https://timelydataflow.github.io/differential-dataflow/chapter_0/chapter_0.html>.  It takes a
+//! collection of `Manages` structs that map from an employee ID to the
+//! employee's manager, and outputs a collection of `SkipLevel` structs that
+//! also include the employee's second-level manager.
+
+use anyhow::Result;
+use clap::Parser;
+use dbsp::{
+    operator::FilterMap,
+    trace::{BatchReader, Cursor},
+    OrdIndexedZSet, OutputHandle, Runtime, Stream,
+};
+use size_of::SizeOf;
+use std::hash::Hash;
+
+type EmployeeID = usize;
+
+/// Indicates that `manager` is the immediate manager of `employee`.
+///
+/// If `manager == employee` then `manager` is the CEO.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, SizeOf)]
+struct Manages {
+    manager: EmployeeID,
+    employee: EmployeeID,
+}
+
+/// Indicates that `manager` is the immediate manager of `employee` and that
+/// `grandmanager` is the immedate manager of `manager`.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, SizeOf)]
+struct SkipLevel {
+    grandmanager: EmployeeID,
+    manager: EmployeeID,
+    employee: EmployeeID,
+}
+
+type Weight = i32;
+type SkipLevels = OrdIndexedZSet<EmployeeID, SkipLevel, Weight>;
+
+fn print_output(output: &OutputHandle<OrdIndexedZSet<usize, SkipLevel, Weight>>) {
+    let output = output.consolidate();
+    let mut cursor = output.cursor();
+    while cursor.key_valid() {
+        while cursor.val_valid() {
+            let weight = cursor.weight();
+            let SkipLevel {
+                grandmanager,
+                manager,
+                employee,
+            } = cursor.val();
+            println!("    ({grandmanager}, {manager}, {employee}) {weight:+}");
+            cursor.step_val();
+        }
+        cursor.step_key();
+    }
+    println!();
+}
+
+#[derive(Debug, Clone, Parser)]
+struct Args {
+    /// Number of employees.
+    #[clap(long, default_value = "10")]
+    size: usize,
+
+    /// Number of threads.
+    #[clap(long, default_value = "2")]
+    threads: usize,
+}
+
+fn main() -> Result<()> {
+    let Args { threads, size } = Args::parse();
+
+    let (mut dbsp, (hmanages, output)) = Runtime::init_circuit(threads, |circuit| {
+        let (manages, hmanages) = circuit.add_input_zset::<Manages, Weight>();
+
+        let manages_by_manager = manages.map_index(|m| (m.manager, m.clone()));
+        let manages_by_employee = manages.map_index(|m| (m.employee, m.clone()));
+
+        // If Manages { manager, employee: common } and Manages { manager: common,
+        // employee } then SkipLevel { grandmanager: manager, manager: common,
+        // employee }.
+        let skiplevels: Stream<_, SkipLevels> = manages_by_employee
+            .join_index::<(), _, _, _, _, _>(&manages_by_manager, |common, m1, m2| {
+                Some((
+                    *common,
```

Comment:
I'm not sure you need to index output by `common`.  Could you just output `OrdZSet<SkipLevel, Weight>` instead of `OrdIndexedZSet<usize, SkipLevel, Weight>`?

Reply:
Oh, thanks for pointing that out. I thought that smelled funny but I didn't understand the difference between `join` and `join_index`. I fixed it with my push just a minute ago.